src/Listener/UpdateCastorListener.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Castor\Listener;
+
+use Castor\Console\Application;
+use JoliCode\PhpOsHelper\OsHelper;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Process\Process;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\HttpClient\Exception\ExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class UpdateCastorListener implements EventSubscriberInterface
+{
+    public function __construct(
+        private readonly CacheItemPoolInterface&CacheInterface $cache,
+        private readonly HttpClientInterface $httpClient,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {
+    }
+
+    public function checkUpdate(ConsoleCommandEvent $event): void
+    {
+        if (class_exists(\RepackedApplication::class)) {
+            return;
+        }
+
+        $command = $event->getCommand();
+        if (!$command) {
+            return;
+        }
+        if ('_complete' === $command->getName()) {
+            return;
+        }
+        $input = $event->getInput();
+        if ($input->hasOption('format') && 'json' === $input->getOption('format')) {
+            return;
+        }
```

Comment:
Very important part !!!! The compile command need to get application description as json.